skimage/feature/peak.py

```diff
@@ -123,6 +157,15 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
 
     out = np.zeros_like(image, dtype=np.bool)
 
+    # no peak for a trivial image
+    if np.all(image == image.flat[0]):
+        if indices is True:
+            return np.empty((0, 2), np.int)
+        else:
+            return out
+
+    threshold_abs = threshold_abs if threshold_abs is not None else image.min()
+
```

Comment:
The comment just below "    # In the case of labels, recursively build and return an output
    # operating on each label separately" seems to be uncorrect to me now. There is no more recursive call and instead, it's a call to ndi on each label. Can you fix this please?